google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -349,19 +398,28 @@ public void run() {
 
           private void maybeRetry() {
             if (attemptsRemaining > 0) {
-              runTransaction(
+              span.addAnnotation("retrying");
+              runTransactionAttempt(
                   transactionCallback,
                   resultFuture,
                   new TransactionOptions(
-                      attemptsRemaining, options.getExecutor(), transaction.getTransactionId()));
+                      attemptsRemaining, options.getExecutor(), transaction.getTransactionId()),
+                  span);
             } else {
+              span.setStatus(
+                  io.opencensus.trace.Status.ABORTED.withDescription("too many retries"));
```

Comment:
you can have the status with your description as static private to avoid allocations.